src/Bridge/Doctrine/MongoDbOdm/Filter/SearchFilter.php

```diff
@@ -0,0 +1,207 @@
+        if (
+            null === $value ||
+            !$this->isPropertyEnabled($property, $resourceClass) ||
+            !$this->isPropertyMapped($property, $resourceClass, true)
+        ) {
+            return;
+        }
+
+        $matchField = $field = $property;
+
+        $associations = [];
+        if ($this->isPropertyNested($property, $resourceClass)) {
+            [$matchField, $field, $associations] = $this->addLookupsForNestedProperty($property, $aggregationBuilder, $resourceClass);
+        }
+        $metadata = $this->getNestedMetadata($resourceClass, $associations);
+
+        $values = $this->normalizeValues((array) $value, $property);
+        if (null === $values) {
+            return;
+        }
+
+        $caseSensitive = true;
+
+        if ($metadata->hasField($field) && !$metadata->hasAssociation($field)) {
+            if ('id' === $field) {
+                $values = array_map([$this, 'getIdFromValue'], $values);
+            }
+
+            if (!$this->hasValidValues($values, $this->getDoctrineFieldType($property, $resourceClass))) {
+                $this->logger->notice('Invalid filter ignored', [
+                    'exception' => new InvalidArgumentException(sprintf('Values for field "%s" are not valid according to the doctrine type.', $field)),
+                ]);
+
+                return;
+            }
+
+            $strategy = $this->properties[$property] ?? self::STRATEGY_EXACT;
+
+            // prefixing the strategy with i makes it case insensitive
+            if (0 === strpos($strategy, 'i')) {
+                $strategy = substr($strategy, 1);
+                $caseSensitive = false;
+            }
+
+            if (1 === \count($values)) {
+                $aggregationBuilder
+                    ->match()
+                    ->field($matchField)
+                    ->equals($this->addEqualityMatchStrategy($strategy, $field, $values[0], $caseSensitive, $metadata));
+
+                return;
+            }
+
+            $inValues = [];
+            foreach ($values as $inValue) {
+                $inValues[] = $this->addEqualityMatchStrategy($strategy, $field, $inValue, $caseSensitive, $metadata);
+            }
+            $aggregationBuilder
+                ->match()
+                ->field($matchField)
+                ->in($inValues);
+        }
+
+        // metadata doesn't have the field, nor an association on the field
+        if (!$metadata->hasAssociation($field)) {
+            return;
+        }
+
+        $values = array_map([$this, 'getIdFromValue'], $values);
+
+        if (!$this->hasValidValues($values, $this->getDoctrineFieldType($property, $resourceClass))) {
+            $this->logger->notice('Invalid filter ignored', [
+                'exception' => new InvalidArgumentException(sprintf('Values for field "%s" are not valid according to the doctrine type.', $property)),
+            ]);
+
+            return;
+        }
+
+        if (1 === \count($values)) {
```

Comment:
Let your database optimise queries for you instead of duplicating logic:
```
> db.nullTest.find({field: { $in: [1] }}).explain()
{
	"queryPlanner" : {
		"plannerVersion" : 1,
		"namespace" : "test.nullTest",
		"indexFilterSet" : false,
		"parsedQuery" : {
			"field" : {
				"$eq" : 1
			}
		},
...
```